test/rain-report.e2e-spec.ts

```diff
@@ -0,0 +1,119 @@
+import { Test, TestingModule } from '@nestjs/testing';
```

Comment:
New file for `/api/data` e2e tests. Including success and error scenarios for `POST /api/data`.